### PR TITLE
Enable deleting messages

### LINKED ISSUE #2775: support deleting messages

### PR DESCRIPTION
**Pull request by jrainville**
Fixes #2775

Do not merge until https://github.com/status-im/status-go/pull/2279 is merged

### CODE REVIEW

File: ui/app/AppLayouts/Chat/components/MessageContextMenu.qml [hunk 1]
```diff
@@ -273,4 +273,33 @@ PopupMenu {
...
+
+    Action {
+        id: deleteMessageAction
+        text: qsTr("Delete message")
+        onTriggered: {
+            if (!appSettings.showDeleteMessageWarning) {
+                return chatsModel.messageView.deleteMessage(messageId)
+            }
+
+            let confirmationDialog = openPopup(genericConfirmationDialog, {
```
Comment (pinebit) [inline ui/app/AppLayouts/Chat/components/MessageContextMenu.qml:285]: why would you need this variable? `confirmationDialog`

File: ui/shared/ConfirmationDialog.qml [hunk 2]
```diff
@@ -61,7 +85,12 @@ ModalPopup {
             anchors.rightMargin: Style.current.smallPadding
             text: confirmationDialog.cancelButtonLabel
             anchors.bottom: parent.bottom
-            onClicked: confirmationDialog.cancelButtonClicked()
+            onClicked: {
+                if (executeConfirm && typeof executeConfirm === "function") {
```
Comment (pinebit) [inline ui/shared/ConfirmationDialog.qml:89]: this is Cancel button, not Confirm, isn't it?

File: ui/shared/ConfirmationDialog.qml [hunk 3]
```diff
@@ -47,11 +65,17 @@ ModalPopup {
...
             type: confirmationDialog.btnType
-            anchors.right: cancelButton.left
-            anchors.rightMargin: Style.current.smallPadding
+            anchors.right: cancelButton.visible ? cancelButton.left : parent.right
+            anchors.rightMargin: cancelButton.visible ? Style.current.smallPadding : 0
             text: confirmationDialog.confirmButtonLabel
             anchors.bottom: parent.bottom
-            onClicked: confirmationDialog.confirmButtonClicked()
+            onClicked: {
+                if (executeConfirm && typeof executeConfirm === "function") {
```
Comment (pinebit) [inline ui/shared/ConfirmationDialog.qml:73]: why do I need this `executeConfirm`? as a user I could simply implement `onConfirmButtonClicked` handler.

File: ui/shared/ConfirmationDialog.qml (hunk 2 above)
Comment (pinebit) [reply ui/shared/ConfirmationDialog.qml:89]: I think the intention here was to use `executeCancel`

**Comment by anastasiyaig** [comment]
@jrainville may i ask u to correct the color of text and hover and icon for delete message button and the typo in `garanteed` wording? Expected:

1. button hover is red
2. button text is read
3. button icon is edit 
4. garanteed is replaced with `guaranteed`

**Comment by anastasiyaig** [comment]
- [ ] when pinning a message and then deleting -> it is not being deleted from PIN modal. Also , the counter is not updated
- [ ] when deleting a pinned message from pin modal -> counter of pinned messages is not updated
- [ ] when deleting pinned message -> it is not being deleted for recipient PIN
- [ ] when deleting a message which is a picture -> the picture is not removed for the recipient

**Comment by jrainville** [comment]
Nice finds both of you. I'll fix right away

**Comment by jrainville** [comment]
@anastasiyaig and @pinebit The issues should be fixed. Thanks again

File: src/app/chat/signal_handling.nim [hunk 4]
```diff
@@ -4,7 +4,7 @@ import
 proc handleSignals(self: ChatController) =
   self.status.events.on(SignalType.Message.event) do(e:Args):
     var data = MessageSignal(e)
-    self.status.chat.update(data.chats, data.messages, data.emojiReactions, data.communities, data.membershipRequests, data.pinnedMessages, data.activityCenterNotification)
+    self.status.chat.update(data.chats, data.messages, data.emojiReactions, data.communities, data.membershipRequests, data.pinnedMessages, data.activityCenterNotification, data.deletedMessages)
```
Comment (richard-ramos) [inline src/app/chat/signal_handling.nim:7]: image

File: ui/app/AppLayouts/Chat/components/MessageContextMenu.qml [hunk 5]
```diff
@@ -273,4 +273,39 @@ PopupMenu {
...
+                                                       confirmationDialog.close()
+                                                       chatsModel.messageView.deleteMessage(messageId)
+                                                   }
+                                               })
+        }
+        icon.source: "../../../img/delete.svg"
+        icon.color: Style.current.danger
+        icon.width: 16
+        icon.height: 16
+        enabled: isCurrentUser
```
Comment (richard-ramos) [inline ui/app/AppLayouts/Chat/components/MessageContextMenu.qml:309]: Probably not all message types can be deleted, but I'm not completely sure. Figma design only shous the delete functionality for text, stickers and images

**Review by richard-ramos — APPROVED** [review]
:medal_sports:

**Comment by anastasiyaig** [comment]
we need somehow to filter out transaction messages from deletion , since deleting them hangs the app

Screenshot 2021-07-19 at 16 36 54

**Comment by anastasiyaig** [comment]
@John-44 @simonricoo should user be able to delete community invitation?

Screenshot 2021-07-19 at 17 37 36

**Comment by jrainville** [comment]
Nice find. It's fixed now. I don't think we should be able to delete community invites, because it would imply that we'd need to remove the right to join, and it's very complex

**Comment by anastasiyaig** [comment]
@jrainville for some reason i am still able to delete community invite in last build

Screenshot 2021-07-20 at 11 46 00

**Comment by jrainville** [comment]
> also i am not able to see mentions in this PR , each time i mention someone , the messages are not there (they appear after relogin)

Do you have a particular reproduction flow for that? I tried it myself and it worked correctly.

These are two new accounts.

One thing I wonder, is did you use the same account that you used in the first time you tested? Because the migration might have changed.

https://user-images.githubusercontent.com/11926403/126338721-7329b2c6-459a-4021-ae11-26c7e6f04a92.mp4

**Comment by anastasiyaig** [comment]
@jrainville ya the app data was old (remaining from previous build )

**Comment by anastasiyaig** [comment]
@jrainville can we also take in account the activity center when deleting messages? 
If i delete mentions in the community channel -> they are still partially shown in activity center of the receiver

Screenshot 2021-07-22 at 13 43 04

**Comment by jrainville** [comment]
>  can we also take in account the activity center when deleting messages?

Yes that would be the goal. I asked for help on the status-go side for that, since it's a tricky situation. I'm waiting from their input to fix it

**Comment by anastasiyaig** [comment]
@jrainville how things are going for status-go side here? I think we need to have it merged before u are out for vacation

**Comment by iurimatias** [comment]
> @jrainville can we also take in account the activity center when deleting messages?
> If i delete mentions in the community channel -> they are still partially shown in activity center of the receiver
> 
> Screenshot 2021-07-22 at 13 43 04

after talking to @jrainville it seems this is something that needs to be implemented in status-go, opened https://github.com/status-im/status-go/issues/2289

**Pull request merged**